src/mysql_conn.erl

```diff
@@ -300,8 +368,10 @@ handle_call({change_user, Username, Password, Options}, From,
     State2 = State1#state{query_cache = empty, stmts = dict:new()},
     case Result of
         #ok{} ->
-            State3 = State2#state{user = Username, password = Password},
-            case execute_on_connect(Queries, Prepares, State3) of
+            State3 = State2#state{user = Username, password = Password,
+                                  database=Database, queries=Queries,
+                                  prepares=Prepares},
```

Comment:
What's happening here, setting database, queries and prepares?

The benefit of using post_connect here would be that we don't need to fetch queries and prepares from the state, since post_connect does that.

Reply:
So this is ok now, yes?

Btw, this showed me an omission I had made earlier, I forgot to put the queries and prepares from the change user call into the state. Not a big deal actually, as they are never used again after this, but still.

Reply:
Ah, good. I didn't remember that change user contains `{change_user, Username, Password, Options}`. I imagined that the same prepares and queries as in the initial connect would be used after change user. I see now.

Also good that it gets saved, for consistency. They're not needed, but at least the old values are erased from the state in this way.

Reply:
Using the same prepares and queries would probably be undesirable, maybe even dangerous, I think =^^= Certainly so if the connections are used in a pool.

Looking at the state record, I see that it carries around a lot of stuff that is not needed after connecting, like the tcp and ssl options, username, password, queries, prepares, ... When I get around to do a cleanup as mentioned above, I think it might be feasible (at a cursory glance) to have in fact _two_ state records, one that carries what is needed for connecting, one that carries what is needed after connecting, start with the not connected state, and replace the gen_server state when "switching" from not connected to connected. I have been doing this a few times in own projects, though often its not worth it, and not sure it is possible here. I'll investigate :)

Reply:
Two states sounds good! Perhaps the state before connecting can just be the list of options received in `init/1`. Then, the check if we're connected is just an `is_list(State)`. I don't know if it's strait-forward but it's an idea at least.

Reply:
I started experimenting with this, but turns out we can't do it like this... Problem is that we need host, port, user, password etc in kill_query. So, there isn't much what could be removed from the state record. Not worth the effort I think :\

Reply:
OK then. Thanks for investigating this though.